File: CS/1528_Shuffle_String.cs
public class Solution {
    public string RestoreString(string s, int[] indices) {
        StringBuilder result = new StringBuilder(s);
        
        int tempI;
        char tempS;
        for(int i = 0; i < indices.Length; i++){
            while(indices[i] != i){
                tempI = indices[indices[i]];
                tempS = result[indices[i]];
                
                result[indices[i]] = result[i];
                indices[indices[i]] = indices[i];
                
                result[i] = tempS;
                indices[i] = tempI;
            }
        }
        
        return result.ToString();
    }
}

File: Python3/1528_Shuffle_String.py
class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        result = ""
        for i in range(0,len(indices)):
            result += s[indices.index(i)]
        return result;
        """
        result = list(s)
        for i in range(0,len(indices)):
            while(i != indices[i]):
                index = indices[i]
                val = result[i]
                indices[i] = indices[index];
                result[i] = result[index]
                indices[index] = index
                result[index] = val
        return ''.join(result)
        """
